### PR TITLE
[13.0][MIG] mail_attach_ existing_attachment : Migration to 13.0

### PR DESCRIPTION
**Pull request by manuelcalerosolis**
This is the migration of mail_attach_ existing_attachment to 13.0

@Tecnativa TT19768

### CODE REVIEW

**Comment by pedrobaeza** [comment]
TT? Everything is red

**Comment by manuelcalerosolis** [comment]
Ready to be reviewed @pedrobaeza @ernestotejeda @chienandalu 
Please review

File: mail_attach_existing_attachment/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with mail_attach_existing_attachment.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Mail Attach Existing Attachment",
+    "summary": "Adding attachment on the object by sending this one",
+    "author": "ACSONE SA/NV, " "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (ernestotejeda) [inline mail_attach_existing_attachment/__manifest__.py:28]:
```suggestion
    "author": "ACSONE SA/NV, Tecnativa, Odoo Community Association (OCA)",
```

File: mail_attach_existing_attachment/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#     mail_attach_existing_attachment is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with mail_attach_existing_attachment.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (ernestotejeda) [inline mail_attach_existing_attachment/__manifest__.py:24]: We can take advantage of this migration to change the header of this file according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L1-L2

File: mail_attach_existing_attachment/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+To use this module, you need to:
+
+* Adding some attachments on an object by creating a new *Log note*
+
+.. figure:: mail_attach_existing_attachment/static/description/attachment.png
```
Comment (ernestotejeda) [inline mail_attach_existing_attachment/readme/USAGE.rst:5]:
```suggestion
.. figure:: ../static/description/attachment.png
```
Change this (and generate README.rst file) to show the image correctly.

File: mail_attach_existing_attachment/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+   :alt: Attachment on purchase order
+
+* Then, by sending the object via email, you can select the attachment added earlier
+
+.. figure:: mail_attach_existing_attachment/static/description/ex_mail_compose_message.png
```
Comment (ernestotejeda) [inline mail_attach_existing_attachment/readme/USAGE.rst:10]:
```suggestion
.. figure:: ../static/description/ex_mail_compose_message.png
```
The same here

File: mail_attach_existing_attachment/wizard/mail_compose_message.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#     mail_attach_existing_attachment is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with mail_attach_existing_attachment.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (ernestotejeda) [inline mail_attach_existing_attachment/wizard/mail_compose_message.py:23]: You can change the header of this file as I commented above

File: mail_attach_existing_attachment/tests/test_mail_attach_existing_attachment.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+#     mail_attach_existing_attachment is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with mail_attach_existing_attachment.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (ernestotejeda) [inline mail_attach_existing_attachment/tests/test_mail_attach_existing_attachment.py:23]: You can change the header of this file as I commented above

**Review by ernestotejeda — CHANGES_REQUESTED** [review]
Thanks @manuelcalerosolis , just a few comments.

**Review by ernestotejeda — COMMENTED** [review]
Why are you deleting some files from mail_activity_done module in this PR ?

Functional review: OK.

File: mail_attach_existing_attachment/wizard/mail_compose_message.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+from odoo import api, fields, models
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = "mail.compose.message"
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super(MailComposeMessage, self).default_get(fields_list)
```
Comment (chienandalu) [inline mail_attach_existing_attachment/wizard/mail_compose_message.py:33]:
```suggestion
        res = super().default_get(fields_list)
```

File: mail_attach_existing_attachment/wizard/mail_compose_message.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    object_attachment_ids = fields.Many2many(
+        comodel_name="ir.attachment",
+        relation="mail_compose_message_ir_attachments_object_rel",
+        column1="wizard_id",
+        column2="attachment_id",
+        string="Object Attachments",
+    )
+
+    def get_mail_values(self, res_ids):
+        res = super(MailComposeMessage, self).get_mail_values(res_ids)
```
Comment (chienandalu) [inline mail_attach_existing_attachment/wizard/mail_compose_message.py:53]:
```suggestion
        res = super().get_mail_values(res_ids)
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
Please, attend @ernestotejeda comments

File: mail_activity_done/static/src/js/mail_activity.js [hunk 9]
```diff
@@ -15,51 +14,54 @@ odoo.define('mail.Activity.done', function(require) {
...
+    var setDelayLabel = function ( activities ) {
         var today = moment().startOf('day');
-        _.each(activities, function(activity) {
+        _.each(activities, function ( activity ) {
             var to_display = '';
             var deadline = moment(activity.date_deadline).startOf('day');
-            var diff = deadline.diff(today, 'days', true); // true means no rounding
-            if(diff === 0){
+            var diff = deadline.diff(today, 'days', true);
+            // True means no rounding
```
Comment (Tardo) [inline mail_activity_done/static/src/js/mail_activity.js:23]: Comment moved

File: mail_activity_done/static/src/js/mail_activity.js [hunk 10]
```diff
@@ -15,51 +14,54 @@ odoo.define('mail.Activity.done', function(require) {
...
         });
         return open_activities;
     };
 
-    var Activity = mail_activity.include({
-        /**
-         * @override
-         * @private
-         */
+    /**
```
Comment (Tardo) [inline mail_activity_done/static/src/js/mail_activity.js:50]: Moved comment

**Review by Tardo — CHANGES_REQUESTED** [review]
Stopped review... please, attend other reviewers comments. Seems that you mixed the history with other module.

**Comment by pedrobaeza** [comment]
Superseded by #594

**Pull request closed without merging**